server/src/main/java/de/uftos/services/StudentGroupService.java

```diff
@@ -142,6 +146,10 @@ public StudentGroupResponseDto update(String id, StudentGroupRequestDto groupReq
 
     List<Grade> grades =
         this.gradeRepository.findAllById(groupRequest.gradeIds());
+    List<Grade> allGrades = this.gradeRepository.findAll();
```

Comment:
I guess these loops are required to remove old grade references and set the new references?

Reply:
Yeah otherwise you get duplicate grades in the student group and duplicate student groups in the grade.
You could compare the grades of the student group in the db and the gradeRequestDto, but i feel like it doesn't make a big difference.